zml/module.zig

```diff
@@ -661,485 +746,6 @@ pub const CompilationContext = struct {
-    const tensor_args = context.tensorFromShapes(ModuleSignature(func).ArgsT, arena, args);
-    // Run in a dedicated thread because compilation relies on `threadlocal`.
-    const f = try asynk.callBlocking(CompilationContext.generateBytecode, .{ context, arena, "main", func, &model, &tensor_args });
-    context._module.getBody().appendOperation(f.mlir_fn);
-
-    const sharding = context._platform.sharding();
-    const mlir_ctx = context._mlir_ctx;
-    context._module.op().setAttributeByName("mhlo.num_replicas", mlir.IntegerAttribute(.i32).init(mlir_ctx, sharding.num_replicas).asAttr());
-    context._module.op().setAttributeByName("mhlo.num_partitions", mlir.IntegerAttribute(.i32).init(mlir_ctx, sharding.num_partitions).asAttr());
-
-    if (context._platform.compilation_options.xla_dump_to) |xla_dump_to| {
-        // Write the mlir to a file. All errors are discarded, since this is for debugging only.
-        if (std.fs.openDirAbsolute(xla_dump_to, .{})) |dir| {
-            const name_attr = context._module.op().getAttributeByName("sym_name").?.as(mlir.StringAttribute).?;
-            const file_name = std.fmt.allocPrint(arena, "{s}.mlir", .{name_attr.value()}) catch name_attr.value();
-            if (dir.createFile(file_name, .{ .truncate = true })) |file| {
-                context._module.op().print(file.writer(), .{ .debug_info = true, .debug_info_pretty_form = false });
-                log.info("Wrote MLIR to {s}/{s}", .{ xla_dump_to, file_name });
-            } else |_| {
-                log.warn("Failed to open {s}", .{file_name});
-            }
-        } else |_| {
-            log.warn("Folder not found {s}", .{xla_dump_to});
-        }
-    }
-
-    const loaded_executable = loadOrCompilePjrtExecutable(arena, context._platform, context._module) catch |err| {
-        log.err(
-            "pjrt-{s} failed to compile following valid MLIR:\n{}\n{}",
-            .{ @tagName(context._platform.target), context._module.op().mlirFormatter(.{}), err },
-        );
-        return err;
-    };
-
-    log.debug("******** ZML generated MLIR ********", .{});
-    log.debug("{}", .{context._module.op().mlirFormatter(.{})});
-
-    if (timer) |*t| {
-        const time_ms = @divFloor(t.lap(), std.time.ns_per_ms);
-        if (time_ms > 1000) log.info("Compilation took {d:.3}s", .{stdx.math.divFloat(f32, time_ms, 1000)});
-    }
-
-    var arena_state_exe = std.heap.ArenaAllocator.init(allocator);
-    const arena_exe = arena_state_exe.allocator();
-
-    return .{
-        .platform = context._platform,
-        .exe = loaded_executable,
-        .model_buffer_count = f.n_model,
-        .args_buffer_count = f.n_args,
-        .result_buffer_count = @intCast(f.res_types.len),
-        .result_buffer_shapes = arena_exe.dupe(Shape, f.res_shapes) catch unreachable,
-        .num_devices = sharding.num_replicas * sharding.num_partitions,
-        ._allocator = arena_state_exe,
-    };
-}
-
-pub fn load(
```

Comment:
I removed `load` which AFAICT wasn't used, and did the same thing than `compileModel`